roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,110 @@
+
+    Check for thinpool usage during a containerized installation
+    """
+
+    name = "docker_storage"
+    tags = ["preflight"]
+
+    max_thinpool_data_usage_percent = 90.0
+    max_thinpool_meta_usage_percent = 90.0
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Only run on hosts that depend on Docker."""
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        is_node = "nodes" in get_var(task_vars, "group_names", default=[])
+        return (super(DockerStorage, cls).is_active(task_vars) and is_containerized) or is_node
+
+    def run(self, tmp, task_vars):
+        try:
+            self.max_thinpool_data_usage_percent = float(get_var(task_vars, "max_thinpool_data_usage_percent",
+                                                                 default=self.max_thinpool_data_usage_percent))
+            self.max_thinpool_meta_usage_percent = float(get_var(task_vars, "max_thinpool_metadata_usage_percent",
+                                                                 default=self.max_thinpool_meta_usage_percent))
+        except ValueError as err:
+            return {
+                "failed": True,
+                "msg": "Unable to convert thinpool data usage limit to float: {}".format(str(err))
+            }
+
+        err_msg = self.check_thinpool_usage(task_vars)
+        if err_msg:
+            return {"failed": True, "msg": err_msg}
+
+        return {}
+
+    def check_thinpool_usage(self, task_vars):
+        lvs = self.get_lvs_data(task_vars)
+        lv_data = self.extract_thinpool_obj(lvs)
+
+        data_percent = self.get_thinpool_data_usage(lv_data)
+        metadata_percent = self.get_thinpool_metadata_usage(lv_data)
+
+        if data_percent > self.max_thinpool_data_usage_percent:
+            msg = "thinpool data usage above maximum threshold of {threshold}%"
+            return msg.format(threshold=self.max_thinpool_data_usage_percent)
+
+        if metadata_percent > self.max_thinpool_meta_usage_percent:
+            msg = "thinpool metadata usage above maximum threshold of {threshold}%"
+            return msg.format(threshold=self.max_thinpool_meta_usage_percent)
+
+        return ""
+
+    def get_lvs_data(self, task_vars):
+        lvs_cmd = "/sbin/lvs --select vg_name=docker --select lv_name=docker-pool --report-format json"
+        result = self.exec_cmd(lvs_cmd, task_vars)
+
+        if result.get("failed", False):
+            msg = "no thinpool usage data returned by the host: {}"
+            raise OpenShiftCheckException(msg.format(result.get("msg", "")))
+
+        try:
+            data_json = json.loads(result.get("stdout", ""))
+        except ValueError as err:
+            raise OpenShiftCheckException("Invalid JSON value returned by lvs command: {}".format(str(err)))
+
+        data = data_json.get("report")
+        if not data:
+            raise OpenShiftCheckException("no thinpool usage data returned by the host.")
```

Comment:
Couple of problems here.

1. This isn't too meaningful to the end user. What were you expecting to get and what did you actually get? What should the user do about it?
2. Exact same error message many times here... it won't be obvious which one it balked on.